api/src/main/java/org/openmrs/Cohort.java

```diff
@@ -164,47 +165,92 @@ public void removeMember(Integer memberId) {
 		removeMember(new Patient(memberId));
 	}
 	
+	/**
+	 * @since 2.1.0
+	 */
 	public void removeMember(Patient patient) {
-		List<CohortMembership> memberToRemoveList = getMembers().stream()
-				.filter(m -> m.getPatient().getPatientId().equals(patient.getPatientId())).collect(Collectors.toList());
-		memberToRemoveList.forEach(m -> m.setEndDate(new Date()));
+		CohortMembership memberToRemove = this.getActiveMembership(patient);
+		if (memberToRemove.getPatient().getPatientId().equals(patient.getPatientId())) {
```

Comment:
You should be changing their return type boolean, i.e they should return true if a membership was actually added or removed otherwise false